src/Components/ArticleCard.jsx

```diff
@@ -1,24 +1,46 @@
 import { ArrowBigUp, ArrowBigDown, Share, MessageCircle } from 'lucide-react';
 import { Link, useNavigate } from 'react-router-dom';
 import { formatDate } from "../../utils/formatting";
+import { updateArticleVotes } from "../api";
+import { useState } from 'react';
 
 const ArticleCard = ({ article }) => {
   const navigate = useNavigate();
+  const [votes, setVotes] = useState(article.votes);
```

Comment:
Setting state with a prop, like `const [votes, setVotes] = useState(article.votes);`, is generally considered bad practice because it creates duplicate sources of truth. Here’s why this can lead to issues such as a lack of synchronisation; If the article.votes prop changes (e.g., the article data is updated), the component’s votes state won't automatically update to reflect this change. This can lead to stale or inconsistent data in the UI.

better to have a `updatedVotes` state and derive the votes to show with `article.votes + updatedVotes`

Reply:
It won't make a huge amount of difference in this example but as a rule can be quite important